DNN Platform/DotNetNuke.Web.MvcPipeline/Controllers/ModuleSettingsViewController.cs

```diff
@@ -0,0 +1,262 @@
+            this.contentSecurityPolicy.ImgSource.AddScheme("data:");
+            this.contentSecurityPolicy.StyleSource.AddInline();
+            return this.View(model);
+        }
+
+        protected override void OnActionExecuting(ActionExecutingContext filterContext)
+        {
+            base.OnActionExecuting(filterContext);
+        }
+
+        private void Initialize()
+        {
+            try
+            {
+                // get ModuleId
+                if (this.Request.QueryString["ModuleId"] != null)
+                {
+                    this.moduleId = int.Parse(this.Request.QueryString["ModuleId"]);
+                }
+
+                if (this.Module.ContentItemId == Null.NullInteger && this.Module.ModuleID != Null.NullInteger)
+                {
+                    // This tab does not have a valid ContentItem
+                    ModuleController.Instance.CreateContentItem(this.Module);
+                    ModuleController.Instance.UpdateModule(this.Module);
+                }
+
+                this.CheckPermissions();
+
+                if (this.Module != null)
+                {
+                    // get module
+                    this.TabModuleId = this.Module.TabModuleID;
+
+                    // get Settings Control
+                    ModuleControlInfo moduleControlInfo = ModuleControlController.GetModuleControlByControlKey("Settings", this.Module.ModuleDefID);
+
+                    if (moduleControlInfo != null)
+                    {
+                        /*
+                        this.control = ModuleControlFactory.LoadSettingsControl(this.Page, this.Module, moduleControlInfo.ControlSrc);
+
+                        var settingsControl = this.control as ISettingsControl;
+                        if (settingsControl != null)
+                        {
+                            this.hlSpecificSettings.Text = Localization.GetString(
+                                "ControlTitle_settings",
+                                settingsControl.LocalResourceFile);
+                            if (string.IsNullOrEmpty(this.hlSpecificSettings.Text))
+                            {
+                                this.hlSpecificSettings.Text =
+                                    string.Format(
+                                        Localization.GetString("ControlTitle_settings", this.LocalResourceFile),
+                                        this.Module.DesktopModule.FriendlyName);
+                            }
+
+                            this.pnlSpecific.Controls.Add(this.control);
+                        }
+                        */
+                    }
+                }
+            }
+            catch (Exception exc)
+            {
+                // Log the exception
+                DotNetNuke.Services.Exceptions.Exceptions.LogException(exc);
+                throw;
+            }
+        }
+
+        private void CheckPermissions()
```

Comment:
Same concern here as my previous comment on the similar code in ModuleSettingsController.cs

Also what is the difference between ModuleSettingscontroller and MOduleSettingsViewController.cs?